src/fish.cpp

```diff
@@ -230,6 +232,7 @@ static int fish_parse_opt(int argc, char **argv, fish_cmd_opts_t *opts) {
                                               {"no-execute", no_argument, NULL, 'n'},
                                               {"profile", required_argument, NULL, 'p'},
                                               {"private", no_argument, NULL, 'P'},
+                                              {"no-config", no_argument, NULL, 'N'},
```

Comment:
Let's not do a short option for this - we have a lot and we have an `n`, let's only add a long option if we will accept this PR. Like the new `--print-rusage-self` option.

Reply:
I wasn't aware the current code allowed for switches without short options.